### PR TITLE
Dev mode changes

### PR DESCRIPTION
**Pull request by k-rtik**
- Simplified POM files, updated dependencies, test uses JUnit 5 now
- Updated integration test so that it doesn't fail if tests run multiple times
- Updated instructions to use `mvn liberty:run` and `mvn liberty:dev`

### CODE REVIEW

**Comment by k-rtik** [comment]
fixed

**Comment by gkwan-ibm** [comment]
@k-rtik 
- [ ] try and change the `version.liberty-maven-plugin` to 3.2 in both pom.xml
- [ ] 2 comment lines in EndpointIT.java are wrapped in the right code pane. Would you format them better?

**Comment by gkwan-ibm** [comment]
hi @Charlotte-Holt , would you review/approve this PR for the changes in the README.adoc?

**Comment by NimG98** [comment]
### Windows end-to-end Test

Error when running integration tests:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running it.io.openliberty.guides.hateoas.EndpointIT
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 1.02 s <<< FAILURE! - in it.io.openliberty.guides.hateoas.EndpointIT
[ERROR] testLinkForInventoryContents  Time elapsed: 0.925 s  <<< ERROR!
java.lang.ClassCastException: org.glassfish.json.JsonObjectBuilderImpl$JsonObjectImpl incompatible with javax.json.JsonArray
        at it.io.openliberty.guides.hateoas.EndpointIT.testLinkForInventoryContents(EndpointIT.java:52)

[ERROR] testLinksForSystem  Time elapsed: 0.055 s  <<< ERROR!
java.lang.ClassCastException: org.glassfish.json.JsonObjectBuilderImpl$JsonObjectImpl incompatible with javax.json.JsonArray
        at it.io.openliberty.guides.hateoas.EndpointIT.testLinksForSystem(EndpointIT.java:99)

[INFO] 
[INFO] Results:
[INFO]
[ERROR] Errors: 
[ERROR]   EndpointIT.testLinkForInventoryContents:52 ClassCast org.glassfish.json.JsonOb...
[ERROR]   EndpointIT.testLinksForSystem:99 ClassCast org.glassfish.json.JsonObjectBuilde...
[INFO]
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0
[INFO]
[ERROR] Integration tests failed: There are test failures.

Please refer to C:\Users\Administrator\Desktop\Guides\guide-rest-hateoas\start\target\failsafe-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
```

File: README.adoc [hunk 1]
```diff
@@ -163,6 +163,9 @@ include::{common-includes}/trywhatyoubuild-end.adoc[]
 
 Navigate to the `start` directory.
 
+[role=command]
+include::{common-includes}/devmode-start.adoc[]
+
 Begin by building your response JSON, which is composed of the `_links` array, as well as the name of
```
Comment (Charlotte-Holt) [inline README.adoc]: I don't think this changed, but I'd suggest updating 
"Begin by building your response JSON, which is composed of the `_links` array, as well as the name of the host machine."
To
"Begin by building your response JSON, which is composed of the `_links` array and the name of
the host machine."

File: README.adoc [hunk 2]
```diff
@@ -301,62 +304,67 @@ and trigger a failure if a change introduces a defect.
...
-Each test method must be marked with the [hotspot=Test]`@Test` annotation. Typically, the execution order of test
-methods is not controlled, but if control is required, you can place methods in a single container test
-method. This container method is then the only one marked with the [hotspot=Test]`@Test` annotation. The following
-[hotspot=testSuite]`testSuite()` contains two test methods, which run in the order they appear:
-
-The test [hotspot=testLinkForInventoryContents]`testLinkForInventoryContents` is responsible for asserting that
-the correct HATEOAS link is created for the inventory contents.
+Each test method must be marked with the [hotspot=Test1 hotspot=Test2]`@Test` annotation. The execution order of test methods
+is controlled by marking them with the [hotspot=Order1 hotspot=Order2]`@Order` annotation. The value passed into the annotation
+denotes the order in which the methods are run.
```
Comment (Charlotte-Holt) [inline README.adoc]: Change
"The value passed into the annotation denotes the order in which the methods are run."
To
"The value that is passed into the annotation denotes the order in which the methods are run."

File: README.adoc [hunk 3]
```diff
@@ -301,62 +304,67 @@ and trigger a failure if a change introduces a defect.
...
 
 The [hotspot=getResponse]`getResponse` helper method reuses the same line of code for retrieving a response from a
 specific URL. This technique helps keep your code neat and organized.
 
-The [hotspot=assertResponse]`assertResponse` method ensures that the response code that you receive is
+The [hotspot=assertResponse]`assertResponse` helper method ensures that the response code that you receive is
 valid (200).
 
-The [hotspot=testLinksForSystem]`testLinksForSystem` test is responsible for asserting that the correct
+The method [hotspot=visitLocalhost]`visitLocalhost` creates a [hotspot=targetResponse]`GET` request to the
```
Comment (Charlotte-Holt) [inline README.adoc]: Change
"The method `visitLocalhost` creates a [hotspot=targetResponse]`GET` request to the `system` service, registering the `localhost` system.
To
"The `visitLocalhost` method creates a `GET` request to the `system` service, which registers the `localhost` system."

File: README.adoc [hunk 4]
```diff
@@ -301,62 +304,67 @@ and trigger a failure if a change introduces a defect.
...
+`system` service, registering the `localhost` system.
+
+The test [hotspot=testLinkForInventoryContents]`testLinkForInventoryContents` is responsible for asserting that
+the correct HATEOAS link is created for the inventory contents.
```
Comment (Charlotte-Holt) [inline README.adoc]: Change
"The test `testLinkForInventoryContents` is responsible for asserting that the correct HATEOAS link is created for the inventory contents."
To
"The `testLinkForInventoryContents` test is responsible for asserting that the correct HATEOAS link is created for the inventory contents.

File: README.adoc [hunk 5]
```diff
@@ -301,62 +304,67 @@ and trigger a failure if a change introduces a defect.
...
+
+Finally, the test [hotspot=testLinksForSystem]`testLinksForSystem` test is responsible for asserting that the correct
```
Comment (Charlotte-Holt) [inline README.adoc]: Change
"Finally, the test `testLinksForSystem` test is responsible for asserting that the correct HATEOAS links are created for the `localhost` system. This method checks for both the `self` link that points to the `inventory` service and the `properties` link that points to the `system` service that is running on the `localhost` system."
To
"Finally, the `testLinksForSystem` test is responsible for asserting that the correct HATEOAS links are created for the `localhost` system. This method checks for both the `self` link that points to the `inventory` service and the `properties` link that points to the `system` service, which is running on the `localhost` system."

**Comment by NimG98** [comment]
- [ ] `isFound = true` should not be in the beginning of the for loop, because it will always be true. And it will never catch when the hostname doesn't exist
    https://github.com/OpenLiberty/guide-rest-hateoas/pull/58/files#diff-ed848138fc31fb2267f78e1e09900d18R90

File: README.adoc [hunk 6]
```diff
@@ -163,14 +163,17 @@ include::{common-includes}/trywhatyoubuild-end.adoc[]
...
+
+Begin by building your response JSON, which is composed of the `_links` array and the name of
 the host machine.
 
 === Linking to inventory contents
 
 As mentioned before, your starting point is an existing simple inventory REST service. 
 
-Take a look at the `request handlers` in the `src/main/java/io/openliberty/guides/microprofile/InventoryResource.java` file.
+Take a look at the `request handlers` in the [file=0]`src/main/java/io/openliberty/guides/microprofile/InventoryResource.java` file.
```
Comment (NimG98) [inline README.adoc]: The hotspot for `src/main/java/io/openliberty/guides/microprofile/InventoryResource.java`
needs to be `[hotspot file=0]`, not `[file=0]`

**Comment by NimG98** [comment]
Format of URL in "Linking to inventory contents" section is all weird: 
image

File: README.adoc [hunk 7]
```diff
@@ -189,18 +192,15 @@ method and integrate it into the [hotspot=PropertiesForHost file=0]`getPropertie
 ----
 
 The contents of your inventory are now under the asterisk (*) wildcard and reside at the following URL:
+http://localhost:9080/inventory/hosts/*[http://localhost:9080/inventory/hosts/*^]
```
Comment (NimG98) [inline README.adoc]: - [ ] Format of URL in "Linking to inventory contents" section is all weird: 
image

File: README.adoc [hunk 8]
```diff
@@ -288,10 +288,11 @@ After the server runs, you can find your new hypermedia-driven `inventory` servi
 At the following URLs, access the `inventory` service that is now driven by hypermedia:
 
 * http://localhost:9080/inventory/hosts[http://localhost:9080/inventory/hosts^]
-* http://localhost:9080/inventory/hosts/localhost[http://localhost:9080/inventory/hosts/{hostname}^]
+* http://localhost:9080/inventory/hosts/localhost[http://localhost:9080/inventory/hosts/localhost}^]
```
Comment (NimG98) [inline README.adoc]: - [ ] `}` at the end of URL should not be here
    image

File: README.adoc [hunk 9]
```diff
@@ -217,7 +214,7 @@ include::finish/src/main/java/io/openliberty/guides/microprofile/InventoryManage
 
 === Linking to each available resource
 
-Take a look at your `InventoryManager` and `InventoryUtil` files.
+Take a look at your [file=0]`InventoryManager` and [file=1]`InventoryUtil` files.
```
Comment (NimG98) [inline README.adoc]: - [ ] file hotspots still not appearing for `InventoryManager` and `InventoryUtil`
    - need to change to `[file=0]` to `[hotspot file=0]`

File: README.adoc [hunk 10]
```diff
@@ -246,7 +243,7 @@ InventoryUtil.java
 include::finish/src/main/java/io/openliberty/guides/microprofile/util/InventoryUtil.java[]
 ----
 
-The [hotspot=buildHostJson file=1]`buildHostJson` helper method builds in the `InventoryUtil` class. It
+The [hotspot=buildHostJson file=1]`buildHostJson` helper method builds in the [file=1]`InventoryUtil` class. It
```
Comment (NimG98) [inline README.adoc]: - [ ] file hotspot still not appearing for `InventoryUtil`
    - need to change to `[file=0]` to `[hotspot file=0]`

**Comment by gkwan-ibm** [comment]
@Charlotte-Holt would you approve this PR?

**Comment by DannyS03** [comment]
- tested/verified on `qa`.
- LGTM.

**Pull request merged**